pkg/operator/controllers/machineset/machineset_controller.go

```diff
@@ -88,11 +102,14 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 		modifiedMachineset.Spec.Replicas = to.Int32Ptr(int32(minSupportedReplicas-replicaCount) + *modifiedMachineset.Spec.Replicas)
 		err := r.client.Update(ctx, modifiedMachineset)
 		if err != nil {
+			r.log.Error(err)
+			conditions.SetControllerDegraded(ctx, r.client, cnds, err)
```

Comment:
My intent here was never to shadow "real" errors encountered in the Reconcile loop. The example on line 112 is one where no execution errors were encountered in the reconcile, so I returned the condition handling error in case it exists. 

This is inconsistent behavior, and I will be updating this implementation to never return the errors from status updates. We should never be in a situation where getting the cluster resource fails (the Reconcile function here would have exited early on line 49 in that case), and would not ever want to change our reconcile functionality if updating controller conditions does not proceed as expected.

Reply:
I will be logging all errors encountered when handling/updating conditions, even if they are thrown away after. Update to the PR incoming.

Reply:
To Kipp's point, I think it's worth logging errors just so they're recorded somewhere, but maybe consider rate limiting them so they don't flood the log?  i.e. if last error occurred less than X seconds ago, skip logging it.